src/containers/Interfaces/AlterForm.js

```diff
@@ -71,18 +71,27 @@ class AlterForm extends Component {
 
   isIntroScreen = () => this.state.activeIndex === 0;
 
-  isLastAlter = () => this.state.activeIndex === this.props.stageNodes.length
+  isLastAlter = () => this.state.activeIndex === this.props.stageNodes.length;
```

Comment:
is `stageNodes` calculated _after_ any stage node filtering has been applied? Just want to make sure on that one.

I initially didn't want automatic skipping for "empty" stages (incase the user wanted to use the information panel to show something to be seen by all participants). Its also less clear what is happening - when clicking a a stage in the menu, you get a prompt if that stage would be skipped based on skip logic, and can choose to force show it. You can't do that with this - its always going to be skipped, by the stage itself.

Reply:
I think that considering we are keeping stage filtering and skip logic separate for now, we shouldn't ever automatically skip a stage. This should be under user control.

Reply:
I think the naming is confusing things here, `shouldSkipAction` is unrelated to the skip logic. This just determines whether or not the stage should block the navigate action or not. I will change the name to make it clearer.

The `stageNodes` prop isn't changed at this is exactly what is displayed to the user when they navigate alters, it uses the `makeNetworkNodesForType` selectors which includes the filter function.

Reply:
FYI the skip logic has been completely removed from the interface/stage components and moved to here: https://github.com/codaco/Network-Canvas/blob/d75099c3d1bf4fa48bad7c95dae50bad911967ee/src/ducks/modules/navigate.js#L33,L61

Reply:
> I initially didn't want automatic skipping for "empty" stages (incase the user wanted to use the information panel to show something to be seen by all participants)

The current implementation maintains this behaviour (of showing info panels with no nodes/edges)

Reply:
Thanks for the explanation. FWIW I wasn't confused about this in the way you think - I was referring to this line because you changed the code following it in a commit called "Skip empty alter and alter edge forms". I wanted to ensure that (1) `stageNodes` took into account filtering for the definition of 'empty', and (2) 'empty' per alter and per alter edge forms were not being automatically skipped (as the title of the commit implies). Seems like the confusion was that I was assuming `shouldSkipAction` was immediately skipping the stage, when it isn't.

Reply:
Yep, should skip action has nothing to do with stage skipping, I've renamed it to `shouldContinue`, which hopefully clarifies intent a little. I also added some explanation of the beforeNext method inline with the code.